File: tests/python/framework.py
#
# Arachne CV
#
# Written by David Hashe and Jacob Brunson
#
#
#
#
# Running multiple webcams might require some hardware configuration to avoid
# 'device full' errors.
#

import ArachneCV as acv

camera0 = acv.Camera(0, [-1, 0, 2], 45, 90, 50);
camera1 = acv.Camera(1, [1, 0, 2], 45, 90, 50);

targets = acv.Targets()

while (True):
    camera0.getFrame()
    camera1.getFrame()

    camera0.warpPerspective()
    camera1.warpPerspective()

    camera0.findTargets()
    camera1.findTargets()

    camera0.showFrame()
    camera1.showFrame()

    targets.merge(camera0.getTargets())
    targets.merge(camera1.getTargets())

    targets.clear()

File: tests/cpp/framework.cpp
/*
 * Arachne CV
 *
 * Written by David Hashe and Jacob Brunson
 *
 *
 *
 * Currently unfinished. Don't actually run this one.
 *
 */

#include <ArachneCV.hpp>

int main()
{
  double cam_coords0[3] = {-1, 0, 2};
  double cam_coords1[3] = {1, 0, 2};

  acv::Camera camera0(0, cam_coords0, 45, 90, 50);
  acv::Camera camera1(1, cam_coords1, 45, 90, 50);

  acv::Targets targets;

  while (true) {
    camera0.getFrame();
    camera1.getFrame();

    // Required for findBalls and findRobots to return correct value, so merge into getFrame?
    camera0.warpPerspective();
    camera1.warpPerspective();

    camera0.findTargets();
    camera1.findTargets();

    targets.merge(camera0.targets);
    targets.merge(camera1.targets);

    targets.clear();
  }

  return 0;
}



